src/server/svr_movejob.c

```diff
@@ -530,14 +530,17 @@ send_job_exec(job *jobp, pbs_net_t hostaddr, int port, struct batch_request *req
 	stream = svr_connect(hostaddr, port, NULL, ToServerDIS, rpp);
 	if (stream < 0) {
 		log_event(PBSEVENT_ERROR, PBS_EVENTCLASS_REQUEST, LOG_WARNING, "", "Could not connect to Mom");
+		pbs_errno = PBSE_NORELYMOM;
 		goto send_err;
 	}
 
 	pmom = tfind2((unsigned long) jobp->ji_qs.ji_un.ji_exect.ji_momaddr,
 		jobp->ji_qs.ji_un.ji_exect.ji_momport,
 		&ipaddrs);
-	if (!pmom || (((mom_svrinfo_t *)(pmom->mi_data))->msr_state & INUSE_DOWN))
+	if (!pmom || (((mom_svrinfo_t *)(pmom->mi_data))->msr_state & INUSE_DOWN)) {
+		pbs_errno = PBSE_NORELYMOM;
```

Comment:
I'm not sure why you set this error number here. If the mom is already not there then why would you want to set this error number and mark the node down in runjob. Also, if the node is already marked down then it does not really help to mark it down again. I could not find a better error number for this case, maybe create a new error number like PBSE_NODEDOWN and purge the subjob upon receiving this error number as well.

Reply:
As seen in the bug description and logs, the error number displayed is "0" which does not provide correct information on the type of failure. hence setting appropriate error number here
```
06/01/2016 17:38:20;0008;Server@pbs-1;Job;42399[1188].pbs-1;Job Run at 
request of Scheduler@pbs-1 on exec_vnode (r311i0n3:ncpus=16) 
06/01/2016 17:38:20;0001;Server@pbs-1;Req;;Could not connect to Mom 
06/01/2016 17:38:20;0008;Server@pbs-1;Job;42399[1188].pbs-1;send of job to 
r311i0n3 failed error = 0 
06/01/2016 17:38:20;0008;Server@pbs-1;Job;42399[1188].pbs-1;Unable to 
Run Job, send to Mom failed 
```

Now after looking at the code of svr_connect() I think You are right that I might be marking the node down again.

I will remove the added call to `mark_node_down()` in `svr_strtjob2()`, but keep the updates to `pbs_errno` for better reporting in the logs.

Reply:
>and purge the subjob upon receiving this error number as well.

Now that `pbs_error` is appropriately set to non-zero, the subjob will get purged down the line at below location in `req_runjob2()`
https://github.com/PBSPro/pbspro/blob/882f33a732d82736aca098d8ee0a7e493287628c/src/server/req_runjob.c#L674-L684